vyxal/parser.py

```diff
@@ -0,0 +1,436 @@
+"""
+File: parser.py
+Description: Once Vyxal programs have been tokenised using lexer.py, the
+next step is to group the tokens into their corresponding structures.
+This is done by treating the tokens as a queue and dequeuing tokens
+until a predicate is matched for structures.
+"""
+
+from __future__ import annotations
+
+import string
+from collections import deque
+
+try:
```

Comment:
This is very sus imo, but what happens if you just do `import lexer` or `import vyxal.lexer`?